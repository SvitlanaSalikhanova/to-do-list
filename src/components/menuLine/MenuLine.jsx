import React from 'react';
import { SHOW_ACTIVE, SHOW_COMPLETED, SHOW_ALL } from '../../FilterConstatnts';
import MenuLineCss from './MenuLine.module.scss';

export default function MenuLine({
    count, changeFilter, deleteCompleted, filter,
}) {
    return (
        <div className={MenuLineCss.menuLine}>
            <div className={MenuLineCss.count}>
                {count}
                {' '}
                item
                {count !== 1 && 's'}
            </div>

            <div className={MenuLineCss.filters}>
                <button
                    type="button"
                    className={
                        filter === SHOW_ALL
                            ? MenuLineCss.filterStateActive
                            : MenuLineCss.filterState
                    }
                    onClick={() => changeFilter(SHOW_ALL)}
                >
                    All
                </button>
                <button
                    type="button"
                    className={
                        filter === SHOW_ACTIVE
                            ? MenuLineCss.filterStateActive
                            : MenuLineCss.filterState
                    }
                    onClick={() => changeFilter(SHOW_ACTIVE)}
                >
                    Active
                </button>
                <button
                    type="button"
                    className={filter === SHOW_COMPLETED
                        ? MenuLineCss.filterStateActive
                        : MenuLineCss.filterState}
                    onClick={() => changeFilter(SHOW_COMPLETED)}
                >
                    Completed
                </button>
            </div>
            <button
                type="button"
                className={MenuLineCss.clearCompleted}
                onClick={deleteCompleted}
            >
                Clear Completed
            </button>

        </div>
    );
}
