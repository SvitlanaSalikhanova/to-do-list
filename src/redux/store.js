import { configureStore } from '@reduxjs/toolkit';
import tasksReducer from './TasksSlice';
import filterReducer from './FilterSlice';

const store = configureStore({
    reducer: {
        tasks: tasksReducer,
        filter: filterReducer,
    },
});

export default store;
