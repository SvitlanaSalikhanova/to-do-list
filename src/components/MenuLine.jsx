import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { setFilter } from '../redux/FilterSlice';
import { clearCompleted } from '../redux/TasksSlice';
import { SHOW_ACTIVE, SHOW_COMPLETED, SHOW_ALL } from '../FilterConstatnts';
import MenuLineCss from '../styles/MenuLine.module.scss';

export default function MenuLine({ count }) {
    const filter = useSelector((state) => state.filter.filter);
    const dispatch = useDispatch();

    return (
        <div className={MenuLineCss.menuLine}>
            <div className={MenuLineCss.count}>
                {count}
                {' '}
                item
                {count !== 1 && 's'}
            </div>

            <div className={MenuLineCss.filters}>
                <button
                    type="button"
                    className={
                        filter === SHOW_ALL
                            ? MenuLineCss.filterStateActive
                            : MenuLineCss.filterState
                    }
                    onClick={() => dispatch(setFilter(SHOW_ALL))}
                >
                    All
                </button>
                <button
                    type="button"
                    className={
                        filter === SHOW_ACTIVE
                            ? MenuLineCss.filterStateActive
                            : MenuLineCss.filterState
                    }
                    onClick={() => dispatch(setFilter(SHOW_ACTIVE))}
                >
                    Active
                </button>
                <button
                    type="button"
                    className={filter === SHOW_COMPLETED
                        ? MenuLineCss.filterStateActive
                        : MenuLineCss.filterState}
                    onClick={() => dispatch(setFilter(SHOW_COMPLETED))}
                >
                    Completed
                </button>
            </div>
            <button
                type="button"
                className={MenuLineCss.clearCompleted}
                onClick={() => dispatch(clearCompleted())}
            >
                Clear Completed
            </button>

        </div>
    );
}
