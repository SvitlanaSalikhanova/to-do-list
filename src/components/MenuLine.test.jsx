import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import MenuLine from './MenuLine';

const onChangeMock = jest.fn();

describe('MenuLine component tests', () => {
    it('MenuLine Matches Snapshot', () => {
        const domTree = renderer.create(
            <Provider store={store}>
                <MenuLine count={1} onChange={onChangeMock} />
            </Provider>,
        ).toJSON();
        expect(domTree).toMatchSnapshot();
    });
});
