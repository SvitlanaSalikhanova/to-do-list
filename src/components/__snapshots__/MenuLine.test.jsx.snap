// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuLine component tests MenuLine Matches Snapshot 1`] = `
<div
  className="menuLine"
>
  <div
    className="count"
  >
    1
     
    item
  </div>
  <div
    className="filters"
  >
    <button
      className="filterStateActive"
      onClick={[Function]}
      type="button"
    >
      All
    </button>
    <button
      className="filterState"
      onClick={[Function]}
      type="button"
    >
      Active
    </button>
    <button
      className="filterState"
      onClick={[Function]}
      type="button"
    >
      Completed
    </button>
  </div>
  <button
    className="clearCompleted"
    onClick={[Function]}
    type="button"
  >
    Clear Completed
  </button>
</div>
`;
